CMake/OtherTests.cmake

```diff
@@ -177,19 +177,20 @@ int main(void) {
   return 0;
 }" HAVE_STRUCT_TIMEVAL)
 
-
-include(CheckCSourceRuns)
-set(CMAKE_REQUIRED_FLAGS)
-if(HAVE_SYS_POLL_H)
-  set(CMAKE_REQUIRED_FLAGS "-DHAVE_SYS_POLL_H")
-endif(HAVE_SYS_POLL_H)
-check_c_source_runs("
+if(NOT ${CMAKE_CROSSCOMPILING})
+  include(CheckCSourceRuns)
+  set(CMAKE_REQUIRED_FLAGS)
+  if(HAVE_SYS_POLL_H)
+    set(CMAKE_REQUIRED_FLAGS "-DHAVE_SYS_POLL_H")
+  endif(HAVE_SYS_POLL_H)
+  check_c_source_runs("
   #ifdef HAVE_SYS_POLL_H
   #  include <sys/poll.h>
   #endif
   int main(void) {
     return poll((void *)0, 0, 10 /*ms*/);
   }" HAVE_POLL_FINE)
+endif()
```

Comment:
Don't know the purpose of this check, but wouldn't it be sufficient to check whether it compiles?